powersimdata/input/changes/plant.py

```diff
@@ -0,0 +1,117 @@
+import copy
+
+from powersimdata.input.transform_grid import TransformGrid
+from powersimdata.utility.distance import find_closest_neighbor
+
+_renewable_resource = {"hydro", "solar", "wind", "wind_offshore"}
```

Comment:
I am wondering if there is not a way to use the `grid.model_immutables.plants.x` where `x` is one of the `set` defined in `powersimdata.network.usa_tamu.constants.plants` (could be network independent in the future). Unfortunately the `renewable_resources` defined there is different than the one defined here. What do you think @BainanXia and @danielolsen?

Reply:
It seems like for the purposes of the change table, this set is 'the resources which have profiles', such that when we try to add a new one, we find the closest profile to grab as the profile for the new plant. It doesn't seem like 'renewable' in the constants really means anything within this repo, but it's used in PostREISE to represent 'resources which have profile AND are curtailable'.

Reply:
Yeah, in change table/grid modification, we care more about whether certain entities have profiles associated with them that we need to deal with, which causes the confusion. Probably we should rename it into something like `resources_with_profiles`?

Reply:
The one in `powersimdata.network.usa_tamu.constants.plants` could then be changed to `curtailable_profile_resources`, or something along those lines.

Reply:
To avoid any issue in PostREISE, we could create a new set in `powersimdata.network.usa_tamu.constants.plants` named `curtailable_profile_resources` and set to `{"hydro", "solar", "wind", "wind_offshore"}` that would be accessed in the change table through `obj.model_immutables.plants.curtailable_profile_resources` in `add_plant`

Reply:
The problem is that `hydro` is a profile resource, but not a curtailable one, so we need two different sets.

Reply:
@jon-hagg, it would be more accurate to rename `_renewable_resource` to `profile_resource`. Can you confirm @BainanXia and @danielolsen.

Reply:
Done (in plant.py only)